### PR TITLE
Docker swarm

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: concourse/resources/dockerSwarm/code/deployBackEnd.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+serviceName=$(echo $buffer | jq '.source.serviceName')
+nodeName=$(echo $buffer | jq '.source.nodeNode')
+swarmMasterIp=$(echo $buffer | jq '.source.swarmMasterIp')
+repository=$(echo $buffer | jq '.source.repository')
+overlay=$(echo $buffer | jq '.source.overlay')
+
+export DOCKER_HOST="tcp://${swarmMasterIp}:2376"
+
+docker run -d -p 9001:9001 --name="${serviceName}" --net=${overlay} --env="constraint:node==${nodeName} ${repository}
```
Comment (csscottlogic) [inline concourse/resources/dockerSwarm/code/deployBackEnd.sh]: Why port 9001?

File: concourse/resources/dockerSwarm/code/deployBackEnd.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+buffer=$(cat)
```
Comment (csscottlogic) [inline concourse/resources/dockerSwarm/code/deployBackEnd.sh]: What does this do?

File: concourse/resources/dockerSwarm/code/deployBackEnd.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+serviceName=$(echo $buffer | jq '.source.serviceName')
+nodeName=$(echo $buffer | jq '.source.nodeNode')
+swarmMasterIp=$(echo $buffer | jq '.source.swarmMasterIp')
+repository=$(echo $buffer | jq '.source.repository')
+overlay=$(echo $buffer | jq '.source.overlay')
+
+export DOCKER_HOST="tcp://${swarmMasterIp}:2376"
```
Comment (csscottlogic) [inline concourse/resources/dockerSwarm/code/deployBackEnd.sh]: Should this be 3376 for the swarm master?

File: concourse/resources/dockerSwarm/code/deployBackEnd.sh (hunk 1 above)
Comment (dwybourn) [inline concourse/resources/dockerSwarm/code/deployBackEnd.sh]: Looks like I forgot to push my changes.

**Comment by csscottlogic** [comment]
:shipit:

**Comment by ianlovell** [comment]
:shipit:

**Pull request merged**